src/actions/profile-view.js

```diff
@@ -123,6 +132,34 @@ export function changeCallTreeSearchString(searchString: string): Action {
   };
 }
 
+function _addAllDescendants(
+  tree: CallTree,
+  newSet: Set<IndexIntoCallNodeTable | null>,
+  callNodeIndex: IndexIntoCallNodeTable
+) {
+  tree.getChildren(callNodeIndex).forEach(childId => {
+    newSet.add(childId);
+    _addAllDescendants(tree, newSet, childId);
+  });
+}
+
+export function expandAllCallNodeDescendants(
+  threadIndex: ThreadIndex,
+  callNodeIndex: IndexIntoCallNodeTable,
+  tree: CallTree,
+  callNodeInfo: CallNodeInfo,
+  expandedCallNodeIndexes: Array<IndexIntoCallNodeTable | null>
+): Action {
```

Comment:
Also maybe you could get `getExpandedCallNodePaths` directly, so that we don't need to convert back and forth between code paths and indexes for the ones that are already expanded.

Reply:
Ah okay, it was because of that guideline I didn't make it a ThunkAction in the first place :) Good to know, I've changed it into such an action now.

If I use `getExpandedCallNodePaths` I cannot use `Set` anymore to get rid of duplicates, can I? I didn't change that part, but please let me know if there's something I'm missing.